dttools/src/copy_tree.c

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (C) 2013- The University of Notre Dame
+ * This software is distributed under the GNU General Public License.
+ * See the file COPYING for details.
+ */
+
+#include <assert.h>
+#include <dirent.h>
+#include <errno.h>
+#include <libgen.h>
+#include <limits.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "copy_tree.h"
+#include "copy_stream.h"
+#include "create_dir.h"
+#include "debug.h"
+#include "path.h"
+#include "xxmalloc.h"
+
+mode_t default_dirmode = S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH;
+
+int copy_dir_real(const char *source, const char *target);
+
+int copy_symlink(const char *source, const char *target) {
+	char linkname[PATH_MAX];
+	ssize_t r;
+
+	assert(source);
+	assert(target);
+
+	if(!access(target, F_OK)) {
+		LDEBUG("%s already exists!\n", target);
```

Comment:
Why LDEBUG? You want notice() of DEBUG here, don't you?
(unless these are temporary while you test the branch)